src/order/zorn.lean

```diff
@@ -268,34 +279,12 @@ let ⟨⟨m, hms, hxm⟩, h⟩ := @zorn_partial_order {m // m ∈ s ∧ x ≤ m}
 theorem zorn_subset {α : Type u} (S : set (set α))
   (h : ∀c ⊆ S, chain (⊆) c → ∃ub ∈ S, ∀ s ∈ c, s ⊆ ub) :
   ∃ m ∈ S, ∀a ∈ S, m ⊆ a → a = m :=
-begin
-  letI : partial_order S := partial_order.lift subtype.val (λ _ _, subtype.ext_val),
-  have : ∀c:set S, @chain S (≤) c → ∃ub, ∀a∈c, a ≤ ub,
-  { intros c hc,
-    rcases h (subtype.val '' c) (image_subset_iff.2 _) _ with ⟨s, sS, hs⟩,
-    { exact ⟨⟨s, sS⟩, λ ⟨x, hx⟩ H, hs _ (mem_image_of_mem _ H)⟩ },
-    { rintro ⟨x, hx⟩ _, exact hx },
-    { rintro _ ⟨x, cx, rfl⟩ _ ⟨y, cy, rfl⟩ xy,
-      exact hc x cx y cy (mt (congr_arg _) xy) } },
-  rcases zorn_partial_order this with ⟨⟨m, mS⟩, hm⟩,
-  exact ⟨m, mS, λ a aS ha, congr_arg subtype.val (hm ⟨a, aS⟩ ha)⟩
-end
+zorn_partial_order₀ S h
 
 theorem zorn_subset₀ {α : Type u} (S : set (set α))
```

Comment:
```suggestion
theorem zorn_subset_nonempty {α : Type u} (S : set (set α))
```
would make more sense to me - I have no idea what the `₀` name means, but comparing the proof here to the proof of `zorn_subset`, the proofs differ by `nonempty` not `₀`.